packages/@webex/plugin-meetings/src/multistream/mediaRequestManager.ts

```diff
@@ -90,14 +100,48 @@ export class MediaRequestManager {
     this.slotsActiveInLastMediaRequest = activeSlots;
   }
 
+  public setDegradationPreferences(degradationPreferences: DegradationPreferences) {
+    this.degradationPreferences = degradationPreferences;
+    this.sendRequests(); // re-send requests after preferences are set
+  }
+
+  private getDegradedClientRequests() {
+    const clientRequests = {...this.clientRequests};
+    const maxFsLimits = [8192, 3600, 920, 240, 60]; // 1080p, 720p, 360p, 180p, 90p
+
+    // reduce max-fs until total macroblocks is below limit
+    for (let i = 0; i < maxFsLimits.length; i += 1) {
+      let totalMacroblocksRequested = 0;
+      Object.values(clientRequests).forEach((mr) => {
+        if (mr.codecInfo) {
+          mr.codecInfo.maxFs = Math.min(
+            mr.codecInfo.maxFs || CODEC_DEFAULTS.h264.maxFs,
+            maxFsLimits[i]
+          );
+          totalMacroblocksRequested += mr.codecInfo.maxFs * mr.receiveSlots.length;
+        }
+      });
+      if (totalMacroblocksRequested <= this.degradationPreferences.maxMacroblocksLimit) {
+        if (i !== 0) {
+          LoggerProxy.logger.warn(
+            `multistream:mediaRequestManager --> too many requests with high max-fs, frame size may be limited`
```

Comment:
instead of saying "may be limited" this log could say "is limited to ${maxFsLimits[i]}", right?

Reply:
Yeah, I kept the message generic since we had discussed that in the future, some requests might be limited while others aren't. But I guess we can change the message when we get there.

Will change it to "will be limited to ${maxFsLimits[i]}".

Reply:
Yeah, I kept the message generic since we had discussed that in the future, some requests might be limited while others aren't. But I guess we can change the message when we get there.

Will change it to "will be limited to ${maxFsLimits[i]}".